q2_moshpit/kraken2/tests/test_classification.py

```diff
@@ -781,5 +913,140 @@ def test_mags(self):
             self.assertFalse(exp_missing & obs)
 
 
+class TestClassifyKraken2MAGs(TestPluginBase):
+    package = "q2_moshpit.kraken2.tests"
+
+    def setUp(self):
+        super().setUp()
+        self.classify_kraken2 = self.plugin.pipelines['classify_kraken2']
+
+    @classmethod
+    def setUpClass(cls):
+        datadir = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), 'data'
+        )
+
+        db_path = os.path.join(datadir, 'simulated-sequences', 'kraken2-db')
+        mags_path = os.path.join(datadir, 'simulated-sequences', 'mags')
+
+        cls.db = Kraken2DBDirectoryFormat(db_path, 'r')
+        cls.samples = MultiFASTADirectoryFormat(mags_path, 'r')
+
+        cls.reports, cls.outputs = _classify_kraken2(cls.samples, cls.db)
+        cls.output_views = cls.outputs.reports.iter_views(pd.DataFrame)
+        cls.report_views = cls.reports.reports.iter_views(pd.DataFrame)
+
+        cls.uuid_to_sample = {
+            'sample2/9231448e-b591-4afc-9d8a-5255b1a24f08': 'ba',
+            'sample2/7797bbd1-4f3c-4482-9828-fa4be13c9977': 'mm',
+            'sample1/5693d0e1-be8e-40ab-9427-94a0ffc62963': 'sa',
+            'sample1/8adb2c2f-bb49-4b1a-a9ac-daf985b35070': 'se',
+        }
+        cls.sample_id_to_ncbi_id = {
+            'ba': {1392},   # bacillus anthracis
+            'mm': {10090},  # mus musculus
+            'sa': {1280},   # staph aureus
+            'se': {1282},   # staph epidermidis
+        }
+
+    def test_formats(self):
+        self.assertIsInstance(self.reports, Kraken2ReportDirectoryFormat)
+        self.assertIsInstance(self.outputs, Kraken2OutputDirectoryFormat)
+        self.reports.validate()
+        self.outputs.validate()
+
+    def test_mags(self):
+        for path, df in self.output_views:
+            mag_id = str(path).rsplit('.output.txt')[0]
+            sample_id = self.uuid_to_sample[mag_id]
+
+            # the expected number of records are in the output
+            self.assertGreater(len(df), 1)
```

Comment:
Here and below, why is the `assertGreater` used instead of `assertEqual`? Do you expect that as Kraken updates are introduced the length of the output tables can increase?

Reply:
That is a good question and it's also something I need to investigate myself 😅 I copied and modified this test as it had been written before. Will look into it ;)